object-filter/src/main/java/org/infinispan/objectfilter/ObjectFilter.java

```diff
@@ -74,9 +74,10 @@ default FilterResult filter(Object value) {
     *
     * @param key   the (optional) key; this can be {@code null} if it is of no interest
     * @param value the instance to test; this is never {@code null}
+    * @param metadata the (optional) metadata; this can be {@code null} if it is of no interest
     * @return a {@link FilterResult} if there is a match or {@code null} otherwise
     */
-   FilterResult filter(Object key, Object value);
+   FilterResult filter(Object key, Object value, Object metadata);
```

Comment:
Why is `Object metadata` used everywhere?

Reply:
`ObjectFilter` is a object filter module class, while `Metadata` is a core class. So I think that the latter is not visible from the former.

Reply:
Here, you can add `infinispan-core`as a dependency. This module is not used by any client.